pkg/api/graph/graph.go

```diff
@@ -142,21 +142,27 @@ func New() Graph {
 	}
 }
 
+// Edges returns all the edges of the graph
 func (g Graph) Edges() []graph.Edge {
 	return g.internal.Edges()
 }
 
+// Len returns the size of the graph
+func (g Graph) Len() int {
```

Comment:
I'm not convinced that length on a graph always means nodes.  Have the caller get the nodes and check len on them.